contracts/misc/LensPeripheryDataProvider.sol

```diff
@@ -0,0 +1,137 @@
+ * a recipient by another user (i.e. Not a mint) and not register them as "following" unless
+ * the recipient explicitly toggles the follow here. 
+ */
+contract LensPeripheryDataProvider {
+    string public constant NAME = 'LensPeripheryDataProvider';
+    bytes32 internal constant EIP712_REVISION_HASH = keccak256('1');
+    bytes32 internal constant EIP712_DOMAIN_TYPEHASH =
+        keccak256(
+            'EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)'
+        );
+    bytes32 internal constant TOGGLE_FOLLOW_WITH_SIG_TYPEHASH =
+        keccak256(
+            'ToggleFollowWithSig(uint256[] profileIds,bool[] enables,uint256 nonce,uint256 deadline)'
+        );
+
+    ILensHub immutable HUB;
+
+    mapping(address => uint256) public sigNonces;
+
+    constructor(ILensHub hub) {
+        HUB = hub;
+    }
+
+    /**
+     * @notice Toggle Follows on the given profiles, emiting toggle event for each FollowNFT.
+     *
+     * NOTE: `profileIds`, `followNFTIds` and `enables` arrays must be of the same length.
+     *
+     * @param profileIds The token ID array of the profiles.
+     * @param enables The array of booleans to enable/disable follows.
+     */
+    function toggleFollow(uint256[] calldata profileIds, bool[] calldata enables) external {
+        _toggleFollow(msg.sender, profileIds, enables);
+    }
+
+    /**
+     * @notice Toggle Follows a given profiles via signature with the specified parameters.
+     *
+     * @param vars A ToggleFollowWithSigData struct containing the regular parameters as well as the signing follower's address
+     * and an EIP712Signature struct.
+     */
+    function toggleFollowWithSig(DataTypes.ToggleFollowWithSigData calldata vars) external {
+        _validateRecoveredAddress(
+            _calculateDigest(
+                keccak256(
+                    abi.encode(
+                        TOGGLE_FOLLOW_WITH_SIG_TYPEHASH,
+                        keccak256(abi.encodePacked(vars.profileIds)),
+                        keccak256(abi.encodePacked(vars.enables)),
+                        sigNonces[vars.follower]++,
+                        vars.sig.deadline
+                    )
+                )
+            ),
+            vars.follower,
+            vars.sig
+        );
+        _toggleFollow(vars.follower, vars.profileIds, vars.enables);
+    }
+
+    function _toggleFollow(
+        address follower,
+        uint256[] calldata profileIds,
+        bool[] calldata enables
+    ) internal {
+        if (profileIds.length != enables.length) revert Errors.ArrayMismatch();
+        for (uint256 i = 0; i < profileIds.length; ++i) {
+            address followNFT = HUB.getFollowNFT(profileIds[i]);
+            if (followNFT == address(0)) revert Errors.FollowInvalid();
+            if (!IERC721Time(address(HUB)).exists(profileIds[i])) revert Errors.TokenDoesNotExist();
```

Comment:
Should `HUB.getFollowNFT(profileId)` perform a  `revert Errors.TokenDoesNotExist()` in case of a non-existent profile? Instead of having to check it manually like this